port/src/main.rs

```diff
@@ -27,93 +23,108 @@ use structopt::StructOpt;
 use uuid::Uuid;
 
 use crate::args::Args;
-use crate::auth::{ApiKey, AuthorizationError, ScopedUser, User, UserDirectory};
 use crate::build::{BuildSystem, FsBuildSystem};
 use crate::deployment::DeploymentSystem;
+use rocket::request::FromRequest;
+use shuttle_service::rocket::Request;
+use shuttle_service::rocket::request::Outcome;
+use rocket::http::Status;
 
 type ApiResult<T, E> = Result<Json<T>, E>;
 
-/// Find user by username and return it's API Key.
-/// if user does not exist create it and update `users` state to `users.toml`.
-/// Finally return user's API Key.
-#[post("/users/<username>")]
-async fn get_or_create_user(
-    user_directory: &State<UserDirectory>,
-    username: String,
-    _admin: Admin,
-) -> Result<ApiKey, AuthorizationError> {
-    user_directory.get_or_create(username)
+struct ControlGuard;
+
+#[async_trait]
+impl<'r> FromRequest<'r> for ControlGuard {
+    type Error = &'static str;
+
+    async fn from_request(request: &'r Request<'_>) -> Outcome<Self, Self::Error> {
+        let state: &'r State<ApiState> = rocket::outcome::try_outcome!(request.guard().await.map_failure(|(s, _)| {
+            (s, "a ControlGuard was used in a route without a state")
+        }));
+        if let Some(auth) = request.headers().get_one("Authorization") {
+            let parts: Vec<&str> = auth.split(' ').collect();
+            if parts.len() != 2 {
+                return Outcome::Failure((Status::Unauthorized, "a request to the control plane was blocked: bad request"));
+            }
+            // unwrap ok because of explicit check above
+            let secret = *parts.get(1).unwrap();
+            if state.deployment_manager.is_authorized(secret) {
```

Comment:
Does this mean we will still need a firewall between projects? Because if each port has its own unique plane secret, then even if users found the internal IPs of other projects then they still won't have the correct plane secret to authenticate their requests?